cdap-common/src/main/java/io/cdap/cdap/common/utils/BatchingConsumer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2018-2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package io.cdap.cdap.common.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class BatchingConsumer<T> implements Consumer<T>, AutoCloseable {
```

Comment:
Add javadocs for the class and public methods. Please note that it must be closed and it's not thread-safe